content/install-keras-environment-from-scratch.md

```diff
@@ -0,0 +1,174 @@
+
+Quick confirmation, you still have time to go back and make changes to your partitioning. Then press the **Continue** button.
+
+![Ubuntu - Write the changes to disks](/img/setup-keras/install/05-ubuntu-erase-disk.png)
+
+Select the city you are located to have the correct timezone. Then press the **Continue** button.
+
+![Ubuntu - Where are you?](/img/setup-keras/install/06-ubuntu-location.png)
+
+Select your keyboard layout. Then press the **Continue** button.
+
+![Ubuntu - Keyboard](/img/setup-keras/install/07-ubuntu-keyboard.png)
+
+Enter your name, the name of your computer, a username and a password. Then press the **Continue** button.
+
+![Ubuntu - User Account](/img/setup-keras/install/08-ubuntu-user-account.png)
+
+Install in progress...
+
+![Ubuntu - Install in progress](/img/setup-keras/install/09-ubuntu-install.png)
+
+You are one reboot away from the Ubuntu desktop. Then press the **Restart Now** button.
+
+![Ubuntu - Restart](/img/setup-keras/install/10-ubuntu-restart.png)
+
+Login using the credentials you've entered in the installation process.
+
+![Ubuntu - Login](/img/setup-keras/install/11-ubuntu-login.png)
+
+Congratulation, you're on the Ubuntu desktop ready to install Keras. You might have some updates to install. Then press the **Install Now** button.
+
+![Ubuntu - Updates](/img/setup-keras/install/12-ubuntu-updates.png)
+
+## Install Keras and its dependencies
+
+First, you must open a **Terminal** and run the script bellow, line by line to install all the requirements of Keras and then Keras itself.
+
+```python
+# Python is installed by default with Ubuntu. 
+# Let's check the current version of Python
+python --version
+```
+
+On my computer I have the version 2.7.6 of python. This will be enough for our dev environment.
+
+![Ubuntu - Python version](/img/setup-keras/terminal/01-python-version.png)
+
+```python
+# Update all the current packages to their latest version
+sudo apt-get update
+
+# A lot of Python packages are available using PIP (Python Package Index)
+# Install the package python-pip
+sudo apt-get -y install python-pip
+
+# Packages needed to build Scipy and Numpy
+sudo apt-get install python2.7-dev build-essential python-dev gfortran
+
+# Other required packages for Scipy and Numpy
+sudo apt-get install libatlas-dev liblapack-dev
+
+# Install numpy and scipy. Those two packages might take a while to install
+sudo pip install numpy
+sudo pip install scipy
+
+# Install git to get the latest version of the next package
+sudo apt-get install git
+
+# Install Theano (backend for Keras)
+# The other option is to TensorFlow (from Google) but we will see that option in a future post
+sudo pip install git+git://github.com/Theano/Theano.git
+
+# Install Keras
+sudo pip install keras
+```
+
+Here you are! You've installed the minimal packages to run your first script using Keras. To make sure everything works, keep your terminal open and enter the following commands:
```

Comment:
This does not cover running things on GPU, which is fundamental to getting things done in deep learning (and is the part of the installation that people usually need help with). We would need to cover:
- installing the Nvidia Cuda drivers
- installing cuDNN
- installing TensorFlow, GPU version
- making sure that Keras is running on GPU (with cuDNN enabled) for both TensorFlow and Theano

Ideally we should make two sections: one about setting up Keras with the Theano backend, one about setting up Keras with the TensorFlow backend.